### PR TITLE
PR: Enable/disable folding guides when cursor enters/exits the left panel

### LINKED ISSUE #12315: Restore code folding for large files
In order to do that we need to do the following:

* Only draw and update the folding panel when the user moves the cursor over it.
* Only draw the visible region in the editor (although that's probably not so necessary if we draw the panel on demand).

@andfoy, please take a look at this one.

### PR DESCRIPTION
**Pull request by andfoy**
## Description of Changes
This PR disables folding guides visibility by default, instead, they will be shown only if the cursor enters the left panel and disabled when the cursor leaves it. We also re-enable the folding support for large files that was disabled previously. 

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [x] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Peek 02-06-2020 14-39

### Issue(s) Resolved

Fixes #12315.

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: @andfoy

### CODE REVIEW

**Comment by goanpeca** [comment]
Nice work @andfoy !

I suggest we always display the folded indicators, (much like VSCode doeS)

image

**Comment by andfoy** [comment]
@goanpeca Here it is:

Peek 02-06-2020 15-21

**Comment by goanpeca** [comment]
Awesome! 🥳

File: spyder/plugins/editor/panels/codefolding.py [hunk 1]
```diff
@@ -248,6 +250,28 @@ def paintEvent(self, event):
...
+                            # it might have been folded by the parent
+                            # editor/document in the case of cloned editor
+                            for deco in self._block_decos:
+                                if deco.block == block:
+                                    # no need to add a deco, just go to the
+                                    # next block
+                                    break
+                            else:
+                                self._add_fold_decoration(block, line_end)
+            return
```
Comment (ccordoba12) [inline spyder/plugins/editor/panels/codefolding.py:274]: The code you added in these lines is very, very similar to the code to draw fold triggers from like 10 lines below. Do you see a way to refactor both and have a single method here and there?

**Comment by andfoy** [comment]
@ccordoba12, this is ready for a final review

**Pull request merged**